server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -821,17 +822,21 @@ void finalizeRecovery(long targetLocalCheckpoint, long trimAboveSeqNo, ActionLis
             final StepListener<Void> handoffListener = new StepListener<>();
             if (request.isPrimaryRelocation()) {
                 logger.trace("performing relocation hand-off");
-                final Consumer<StepListener> forceSegRepConsumer = shard.indexSettings().isSegRepEnabled()
+                final IndexSettings indexSettings = shard.indexSettings();
+                final Consumer<StepListener> forceSegRepConsumer = indexSettings.isSegRepEnabled()
                     ? recoveryTarget::forceSegmentFileSync
                     : res -> res.onResponse(null);
+                final boolean syncTranslog = indexSettings.isRemoteTranslogStoreEnabled()
```

Comment:
Why don't we have the same check in `IndexShard.relocated()` and not pass it as an extra parameter?

Reply:
We can do that.